### ISSUE TITLE
about'exp_mask'

### ISSUE THREAD

**Issue by JGGGBang** [fisherliaoc/SfMLearner-WCE#3]
exp_mask3 = torch.cat((gumbel_softmax(x3[:,:2])[:,:1], gumbel_softmax(x3[:,2:])[:,:1]), 1)
        exp_mask2 = torch.cat((gumbel_softmax(x2[:,:2])[:,:1], gumbel_softmax(x2[:,2:])[:,:1]), 1)
        exp_mask1 = torch.cat((gumbel_softmax(x1[:,:2])[:,:1], gumbel_softmax(x1[:,2:])[:,:1]), 1)
**************************************************************
After reading the paper, I became confused about PM and DM
How should I use these three quantities after obtaining them
What do gumbel softmax (x1 [:,: 2]) [:,: 1] and gumbel softmax (x1 [:, 2:]) [:,: 1] represent respectively
